client/tests/integration_test.rs

```diff
@@ -747,6 +774,48 @@ mod put_deploy {
     }
 }
 
+mod rate_limit {
```

Comment:
I'm not sure this test really makes sense here.  This is proving that we can rate-limit our mock server, not our production node, and it doesn't test the client in any way (unless I'm misunderstanding it?)

Reply:
I thought it was worthwhile to add here because while it's not a real test of our production code, it does actually instantiate a client and make http requests against it.

In doing that I found a bug in the client where if it's executed within a tokio runtime it can overwhelm the executor and block all task processing threads.

Reply:
Hmm - well, it does instantiate a client, but it's not _testing_ the client (at least not any more than the existing tests do).

And I wouldn't agree that that's a bug - it's just a misuse of the client inside the test really?  There wasn't a bugfix applied to the client as a result of this.

This feels more like it belongs as an example than a test.  However, I don't want to block this PR over this.

Could we at least reduce the test run time down from 10 seconds?  That seems a bit excessive to me.